src/vite/vite.config.ts

```diff
@@ -14,9 +14,24 @@ import { virtualRoutes } from './plugins/virtual-routes.js'
 import { virtualStyles } from './plugins/virtual-styles.js'
 import { resolveVocsConfig } from './utils/resolveVocsConfig.js'
 
+const hasReactPlugin = async (plugins: ReadonlyArray<PluginOption>) => {
+  for await (const plugin of plugins) {
+    if (
+      plugin &&
+      ((!Array.isArray(plugin) && plugin.name === 'vite:react-babel') ||
+        (Array.isArray(plugin) && (await hasReactPlugin(plugin))))
```

Comment:
A "plugin" can be either falsy, a plugin, an array of plugins, or a promise returning any of those values. TypeScript requires the double `Array.isArray(…)` check because it can't do the refinement otherwise.